lib/quiz/card_list.dart

```diff
@@ -0,0 +1,303 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:maui/components/quiz_button.dart';
+
+enum OptionCategory { oneAtATime, many, pair }
+
+const Map<String, dynamic> testMap = {
+  'question': 'Match the following according to the habitat of each animal',
+  'choices': ["abc", "def", "stickers/giraffe/giraffe.png", "lmn"],
+  'answer': ['a', 'b', 'c', 'd'],
+  'correct': null,
+  'total': null,
+  'correctSequenceChoices': null,
+  'choicesRightOrWrong': null
+};
+
+class CardList extends StatefulWidget {
+  final Map<String, dynamic> input;
+  final OptionCategory optionsType;
+  final Function onEnd;
+
+  const CardList(
+      {Key key,
+      this.input = testMap,
+      this.optionsType = OptionCategory.many,
+      this.onEnd})
+      : super(key: key);
+
+  @override
+  State createState() => new CardListState();
+}
+
+class CardListState extends State<CardList> {
+  List<String> clicked = [],
```

Comment:
Do you need clicked to be a List of String or is it better to have it be a List of bool